File: Day25/RotateArray.java
import java.util.*;
public class RotateArray {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;
        for (int i = 0, j = n - 1; i < j; i++, j--) {
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        for (int i = 0, j = k - 1; i < j; i++, j--) {
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        for (int i = k, j = n - 1; i < j; i++, j--) {
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter array elements separated by space: ");
        String[] parts = sc.nextLine().split(" ");
        int[] nums = new int[parts.length];
        for (int i = 0; i < parts.length; i++) nums[i] = Integer.parseInt(parts[i]);
        System.out.print("Enter k: ");
        int k = Integer.parseInt(sc.nextLine());
        new RotateArray().rotate(nums, k);
        System.out.println("Rotated array: " + Arrays.toString(nums));
    }
}

File: Day25/RotateArray.py
class Solution(object):
    def rotate(self, nums, k):
        n = len(nums)
        k = k % n
        l1 = nums[n - k:]
        l2 = nums[:n - k]
        l1.extend(l2)
        nums[:] = l1

if __name__ == "__main__":
    nums = list(map(int, input("Enter array elements separated by space: ").split()))
    k = int(input("Enter k: "))
    Solution().rotate(nums, k)
    print("Rotated array:", nums)
